components/apimgt/org.wso2.carbon.apimgt.rest.api.authenticator/src/main/java/org/wso2/carbon/apimgt/rest/api/authenticator/AuthenticatorAPI.java

```diff
@@ -319,6 +318,16 @@ public Response callback(@Context Request request, @PathParam("appName") String
                     log.debug("Set cookies for " + appName + " application.");
                 }
                 // Redirect to the store/apis page (redirect URL)
+                String uiServiceUrl;
+                //The first host in the list "allowedHosts" is the host of UI-Service
+                String uiServiceHost = APIMConfigurationService.getInstance().getApimConfigurations()
+                        .getEnvironmentConfigurations().getAllowedHosts().get(0);
+                if (uiServiceHost.isEmpty()) {
+                    uiServiceUrl = ServiceReferenceHolder.getInstance().getAPIMAppConfiguration().getApimBaseUrl();
```

Comment:
So why we need host and base url both? When host is provided base url is ommitted even it has specified. We need to be consistent and use single url. Looking at two places will cause issues. Let's add a debug log as well.

Reply:
Done with https://github.com/wso2/carbon-apimgt/pull/4896/commits/131693463cb521f3aa19ae56aab16f2c65550c2c